File: src/main/java/net/ccbluex/liquidbounce/ui/client/gui/clickgui/style/styles/tenacity/SideGui/GuiPanel.java
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge by LiquidBounce.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.gui.clickgui.style.styles.tenacity.SideGui;

import net.minecraft.client.Minecraft;

public abstract class GuiPanel {

    final Minecraft mc = Minecraft.getMinecraft();
    public float rectWidth, rectHeight;

    abstract public void initGui();

    abstract public void keyTyped(char typedChar, int keyCode);

    abstract public void drawScreen(int mouseX, int mouseY, float partialTicks, int alpha);

    abstract public void mouseClicked(int mouseX, int mouseY, int button);

    abstract public void mouseReleased(int mouseX, int mouseY, int button);

}

File: src/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/style/styles/fdpdropdown/SideGui/GuiPanel.kt
/*
 * FDPClient Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/SkidderMC/FDPClient/
 */
package net.ccbluex.liquidbounce.ui.client.clickgui.style.styles.fdpdropdown.SideGui

abstract class GuiPanel {
    var rectWidth: Float = 0f
    var rectHeight: Float = 0f

    abstract fun initGui()

    abstract fun keyTyped(typedChar: Char, keyCode: Int)

    abstract fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float, alpha: Int)

    abstract fun mouseClicked(mouseX: Int, mouseY: Int, button: Int)

    abstract fun mouseReleased(mouseX: Int, mouseY: Int, button: Int)
}